File: android/src/main/kotlin/com/neverjp/background_task/lib/BgEventStreamHandler.kt
package com.neverjp.background_task.lib

import io.flutter.plugin.common.EventChannel

class BgEventStreamHandler:  EventChannel.StreamHandler  {
    companion object {
        var eventSink: EventChannel.EventSink? = null
    }

    override fun onListen(arguments: Any?, sink: EventChannel.EventSink?) {
        eventSink = sink
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}

File: ios/background_task/Sources/background_task/lib/BgEventStreamHandler.swift
//
//  BgEventStreamHandler.swift
//  background_task
//
//  Created by 中川祥平 on 2024/01/19.
//

import Flutter

final class BgEventStreamHandler: NSObject, FlutterStreamHandler {
    
    static var eventSink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        Self.eventSink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        Self.eventSink = nil
        return nil
    }
}
